packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart

```diff
@@ -755,3 +763,59 @@ class DebuggerController extends DisposableController
     return positions;
   }
 }
+
+class ScriptCache {
+  ScriptCache();
+
+  Map<String, Script> _scripts = {};
+  final Map<String, Future<Script>> _inProgress = {};
+
+  /// Return a cached [Script] for the given [ScriptRef], returning null
+  /// if there is no cached [Script].
+  Script getScriptCached(ScriptRef scriptRef) {
+    // Check to see if this ScriptRef is really a Script.
+    if (scriptRef is Script) {
+      if (_scripts[scriptRef.id] == null) {
+        _scripts[scriptRef.id] = scriptRef;
+      }
+
```

Comment:
nit: not your fault, but I don't think these new lines add much value. Let's remove them to make this method more dense

Reply:
Agreed, thinking about this, it's not likely that we'll have gotten a `Script` object without having already gone through this caching code; removed.